server/routes.go

```diff
@@ -1010,6 +1010,44 @@ func streamResponse(c *gin.Context, ch chan any) {
 	})
 }
 
+// contextLimitPrompt builds a prompt to send to a running model while trimming it to fit the template into the max context length
+func contextLimitPrompt(ctx context.Context, prompts []string, ctxLen int) (string, error) {
```

Comment:
This should try to preserve the system prompt as well

Reply:
Yes the original system prompt should be kept

Reply:
If there's multiple, probably the most recent